### PR TITLE
Feature/syal 576 syal 577

### PR DESCRIPTION
**Pull request by HieuLT-AVISPL**
This is the ongoing code. Please help to review it!

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        <dependency>
+            <groupId>net.java.dev.jna</groupId>
+            <artifactId>jna</artifactId>
+            <scope>provided</scope>
+            <version>5.5.0</version>
+        </dependency>
+        <dependency>
+            <groupId>net.java.dev.jna</groupId>
+            <artifactId>jna-platform</artifactId>
+            <version>5.5.0</version>
```
Comment (DuyN-AVISPL) [inline pom.xml:78]: please add scope for all the provided dependencies
Please organize dependencies order like provided first, then test dependencies, etc

File: src/main/java/com/symphony/dal/communicator/axis/AxisCommunicator.java [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+import com.google.gson.Gson;
+import com.symphony.dal.communicator.axis.common.*;
+import com.symphony.dal.communicator.axis.dto.DeviceInfo;
+import com.symphony.dal.communicator.axis.dto.SchemaVersionStatus;
+import com.symphony.dal.communicator.axis.dto.VideoOutput;
+import com.symphony.dal.communicator.axis.dto.metric.Device;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+
+import java.util.*;
```
Comment (DuyN-AVISPL) [inline src/main/java/com/symphony/dal/communicator/axis/AxisCommunicator.java:23]: Please import what we're using

Comment (HieuLT-AVISPL) [reply src/main/java/com/symphony/dal/communicator/axis/AxisCommunicator.java]: I fixed it

File: pom.xml (hunk 1 above)
Comment (HieuLT-AVISPL) [reply pom.xml]: I fixed it

**Review by DuyN-AVISPL — APPROVED** [review]

File: src/main/java/com/symphony/dal/communicator/axis/AxisCommunicator.java [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+                boolean videoSource = responseData.getSuccess().getVideoSourcesSuccess().getVideoSource().isActive();
+                if (videoSource)
+                    return AxisConstant.ACTIVE;
+                return AxisConstant.INACTIVE;
+            } else {
+                failedMonitor.put(AxisMonitoringMetric.VIDEO_SOURCE.toString(), responseData.getError().getGeneralError().getErrorDescription());
+                return AxisConstant.NONE;
+            }
+        } catch (Exception e) {
+            logger.error("get video source error: " + e.getMessage());
```
[the remark this answers is not in the document]
Comment (HieuLT-AVISPL) [reply src/main/java/com/symphony/dal/communicator/axis/AxisCommunicator.java]: I fixed it

File: src/main/java/com/symphony/dal/communicator/axis/AxisCommunicator.java [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+    }
+
+    /**
+     * Retrieve device information
+     *
+     * @param stats list statistics properry
+     * @return
+     * @throws Exception if any error occurs
+     */
+    private void retrieveInfoDevice(Map<String, String> stats) {
```
[the remark this answers is not in the document]
Comment (HieuLT-AVISPL) [reply src/main/java/com/symphony/dal/communicator/axis/AxisCommunicator.java]: I fixed it

File: src/main/java/com/symphony/dal/communicator/axis/common/AxisControllingMetric.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (c) 2015-2021 AVI-SPL, Inc. All Rights Reserved.
+ */
+package com.symphony.dal.communicator.axis.common;
+
+public enum AxisControllingMetric {
```
[the remark this answers is not in the document]
Comment (HieuLT-AVISPL) [reply src/main/java/com/symphony/dal/communicator/axis/common/AxisControllingMetric.java:6]: We have a plan to use it.
I fixed it.

File: src/main/java/com/symphony/dal/communicator/axis/common/AxisMonitoringMetric.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2015-2021 AVI-SPL, Inc. All Rights Reserved.
+ */
+package com.symphony.dal.communicator.axis.common;
+
+public enum AxisMonitoringMetric {
```
[the remark this answers is not in the document]
Comment (HieuLT-AVISPL) [reply src/main/java/com/symphony/dal/communicator/axis/common/AxisMonitoringMetric.java:6]: I fixed it

**Pull request merged**